### PR TITLE
Gm focus

### PR DESCRIPTION
**Pull request by mgalbier**
- Add new field managerName_ to TreeManager
- Set manager name for old-/newVersionManager in DiffManager
- Add new attribute managerName to CommentedNode 
- Improve commentedNode loading
- Correct setting of nodeId attribute
- Correct description string for CCodeReview  
- Move creation of overlays after loading of comments to CodeReviewManager
- Initial version of focus feature
- Only execute focus if focusInformation was found
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71662884%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71663068%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71663188%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71664334%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71664384%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71665101%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71665197%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71666295%22%2C%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71669060%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%2095afc27a6750d4817fcbb272cfcd880e534f361f%20VersionControlUI/src/DiffManager.cpp%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71663068%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22So%20the%20manager%20name%20includes%20both%20the%20project%20and%20the%20version%20name%3F%20Why%20is%20that%20necessary%3F%20The%20project%20name%20is%20already%20in%20%60manager-%3Ename%28%29%60%3F%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A08%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Another%20thing%20is%20that%20this%20commit%20should%20have%20ideally%20included%20the%20code%20where%20you%20use%20the%20%60managerName%60%2C%20cause%20right%20now%20you%27re%20just%20setting%20a%20field%20and%20it%20doesn%27t%20become%20clear%20why%20this%20is%20necessary.%20I%27m%20guess%20further%20commits%20will%20make%20this%20clear.%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A09%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20VersionControlUI/src/DiffManager.cpp%3AL94-105%22%7D%2C%20%22Pull%20b55bd7f6c968c8bbda0ed433cedd95faf4ab9490%20ModelBase/src/model/TreeManager.h%2034%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71662884%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20don%27t%20like%20the%20name%20of%20this%20field.%20Now%20manager%20has%20the%20%60TreeManager%3A%3Aname%60%20and%20%60TreeManager%3A%3AmanagerName%60%20fields.%20These%20are%20totally%20identical%2C%20unless%20one%20reads%20the%20comments.%20We%20should%20call%20it%20something%20else.%20How%20about%20%60revisionName%60%20or%20is%20there%20an%20even%20more%20precise%20name%3F%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A06%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20could%20also%20call%20it%20%60revisionName%60%20and%20only%20distinguish%20between%20new%20and%20old.%20That%20would%20also%20work.%5Cr%5CnBTW%3A%20is%20%60TreeManager%3A%3Aname%60%20not%20also%20a%20bad%20name%2C%20since%20it%20is%20not%20the%20name%20of%20the%20manager%20but%20the%20name%20of%20the%20tree%20that%20is%20managed%3F%20%28At%20least%20if%20I%20understood%20correctly%29%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A24%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17177211%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mgalbier%22%7D%7D%2C%20%7B%22body%22%3A%20%22yes%2C%20%60name%60%20is%20also%20not%20ideal%2C%20but%20it%27s%20also%20not%20that%20bad%2C%20since%20a%20manager%27s%20name%20is%20supposed%20to%20be%20the%20name%20of%20the%20top-level%20root%20node.%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A33%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ModelBase/src/model/TreeManager.h%3AL506-515%22%7D%2C%20%22Pull%20d3d47864b103eb32eacaa82fb9996a43c5a694ef%20CodeReview/src/nodes/CommentedNode.h%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71664334%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Dont%27%20use%20just%20%60ATTRIBUTE%60%2C%20instead%20use%20%60ATTRIBUTE_VALUE%60%20as%20in%20for%20example%20is%20done%20in%20%60Comments%3A%3ACommentFreeNode%60.%20That%20makes%20it%20so%20that%20you%20don%27t%20have%20to%20always%20use%20%60set%60%20and%20%60get%60%20but%20can%20directly%20work%20with%20the%20value.%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A18%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22You%20should%20also%20do%20this%20for%20the%20%60offsetX/Y%60%20below.%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A19%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Oh%20nice%2C%20did%20not%20know%20about%20that.%20I%27ll%20change%20it.%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A25%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/17177211%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mgalbier%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20CodeReview/src/nodes/CommentedNode.h%3AL48-61%22%7D%2C%20%22Pull%20856fbe19d9c04ad42875b218cbcb01513b52a14e%20CodeReview/src/nodes/CommentedNode.cpp%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/433%23discussion_r71669060%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60isEmpty%60%22%2C%20%22created_at%22%3A%20%222016-07-21T08%3A53%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1152976%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20CodeReview/src/nodes/CommentedNode.cpp%3AL50-62%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull b55bd7f6c968c8bbda0ed433cedd95faf4ab9490 ModelBase/src/model/TreeManager.h 34'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/433#discussion_r71662884'>File: ModelBase/src/model/TreeManager.h:L506-515</a></b>
- <a href='https://github.com/dimitar-asenov'></a> I don't like the name of this field. Now manager has the `TreeManager::name` and `TreeManager::managerName` fields. These are totally identical, unless one reads the comments. We should call it something else. How about `revisionName` or is there an even more precise name?
- <a href='https://github.com/mgalbier'></a> I could also call it `revisionName` and only distinguish between new and old. That would also work.
  BTW: is `TreeManager::name` not also a bad name, since it is not the name of the manager but the name of the tree that is managed? (At least if I understood correctly)
- <a href='https://github.com/dimitar-asenov'></a> yes, `name` is also not ideal, but it's also not that bad, since a manager's name is supposed to be the name of the top-level root node.
- [ ] <a href='#crh-comment-Pull 95afc27a6750d4817fcbb272cfcd880e534f361f VersionControlUI/src/DiffManager.cpp 4'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/433#discussion_r71663068'>File: VersionControlUI/src/DiffManager.cpp:L94-105</a></b>
- <a href='https://github.com/dimitar-asenov'></a> So the manager name includes both the project and the version name? Why is that necessary? The project name is already in `manager->name()`?
- <a href='https://github.com/dimitar-asenov'></a> Another thing is that this commit should have ideally included the code where you use the `managerName`, cause right now you're just setting a field and it doesn't become clear why this is necessary. I'm guess further commits will make this clear.
- [ ] <a href='#crh-comment-Pull d3d47864b103eb32eacaa82fb9996a43c5a694ef CodeReview/src/nodes/CommentedNode.h 20'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/433#discussion_r71664334'>File: CodeReview/src/nodes/CommentedNode.h:L48-61</a></b>
- <a href='https://github.com/dimitar-asenov'></a> Dont' use just `ATTRIBUTE`, instead use `ATTRIBUTE_VALUE` as in for example is done in `Comments::CommentFreeNode`. That makes it so that you don't have to always use `set` and `get` but can directly work with the value.
- <a href='https://github.com/dimitar-asenov'></a> You should also do this for the `offsetX/Y` below.
- <a href='https://github.com/mgalbier'></a> Oh nice, did not know about that. I'll change it.
- [ ] <a href='#crh-comment-Pull 856fbe19d9c04ad42875b218cbcb01513b52a14e CodeReview/src/nodes/CommentedNode.cpp 6'></a> &nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/433#discussion_r71669060'>File: CodeReview/src/nodes/CommentedNode.cpp:L50-62</a></b>
- <a href='https://github.com/dimitar-asenov'></a> `isEmpty`

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/433?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/433?approve=1'></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/433'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dimitar-asenov** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/433'>Refresh</a>.<a href='#crh-update-none'></a>

File: ModelBase/src/model/TreeManager.h [hunk 1]
```diff
@@ -490,6 +506,9 @@ inline Node* TreeManager::root(){ return root_; }
 inline QString TreeManager::name() { return name_; }
 inline void TreeManager::setName(const QString& name) { name_ = name; }
 
+inline QString TreeManager::managerName() { return managerName_; }
```
Comment (dimitar-asenov) [inline ModelBase/src/model/TreeManager.h:509]: I don't like the name of this field. Now manager has the `TreeManager::name` and `TreeManager::managerName` fields. These are totally identical, unless one reads the comments. We should call it something else. How about `revisionName` or is there an even more precise name?

File: VersionControlUI/src/DiffManager.cpp [hunk 2]
```diff
@@ -94,9 +94,11 @@ DiffSetup DiffManager::initializeDiffPrerequisites(QString oldVersion, QString n
 
 	// load newer version
 	diffSetup.newVersionManager_ = createTreeManagerFromVersion(diffSetup.repository_, diffSetup.newVersion_);
+	diffSetup.newVersionManager_->setManagerName(project_+"#newVersion");
```
Comment (dimitar-asenov) [inline VersionControlUI/src/DiffManager.cpp:97]: So the manager name includes both the project and the version name? Why is that necessary? The project name is already in `manager->name()`?

Comment (dimitar-asenov) [inline VersionControlUI/src/DiffManager.cpp:97]: Another thing is that this commit should have ideally included the code where you use the `managerName`, cause right now you're just setting a field and it doesn't become clear why this is necessary. I'm guess further commits will make this clear.

File: CodeReview/src/nodes/CommentedNode.h [hunk 3]
```diff
@@ -45,12 +48,13 @@ class CODEREVIEW_API CommentedNode : public Super<Model::CompositeNode>
 	COMPOSITENODE_DECLARE_STANDARD_METHODS(CommentedNode)
 
 	ATTRIBUTE(Model::Text, nodeId, setNodeId)
+	ATTRIBUTE(Model::Text, managerName, setManagerName)
```
Comment (dimitar-asenov) [inline CodeReview/src/nodes/CommentedNode.h:51]: Dont' use just `ATTRIBUTE`, instead use `ATTRIBUTE_VALUE` as in for example is done in `Comments::CommentFreeNode`. That makes it so that you don't have to always use `set` and `get` but can directly work with the value.

Comment (dimitar-asenov) [inline CodeReview/src/nodes/CommentedNode.h:51]: You should also do this for the `offsetX/Y` below.

File: ModelBase/src/model/TreeManager.h (hunk 1 above)
Comment (mgalbier) [inline ModelBase/src/model/TreeManager.h:509]: I could also call it `revisionName` and only distinguish between new and old. That would also work.
BTW: is `TreeManager::name` not also a bad name, since it is not the name of the manager but the name of the tree that is managed? (At least if I understood correctly)

File: CodeReview/src/nodes/CommentedNode.h (hunk 3 above)
Comment (mgalbier) [inline CodeReview/src/nodes/CommentedNode.h:51]: Oh nice, did not know about that. I'll change it.

File: ModelBase/src/model/TreeManager.h (hunk 1 above)
Comment (dimitar-asenov) [inline ModelBase/src/model/TreeManager.h:509]: yes, `name` is also not ideal, but it's also not that bad, since a manager's name is supposed to be the name of the top-level root node.

File: CodeReview/src/nodes/CommentedNode.cpp [hunk 4]
```diff
@@ -50,4 +50,12 @@ CommentedNode::CommentedNode(QString associatedNodeId, QString nodeManagerName,
 	offsetY()->set(offset.y());
 }
 
+bool CommentedNode::parseReviewComments(FocusInformation& focusInformation)
+{
+	if (auto firstReviewComment = reviewComments()->first())
```
Comment (dimitar-asenov) [inline CodeReview/src/nodes/CommentedNode.cpp:55]: `isEmpty`
